python/src/aiconfig/editor/client/src/components/SettingsPropertyRenderer.tsx

```diff
@@ -90,7 +91,7 @@ export default function SettingsPropertyRenderer({
   // Used in the case the property is an array
   // TODO: Should initialize with values from settings if available
   const [itemControls, setItemControls] = useState<JSX.Element[]>([]);
-  const itemValues = useRef(new Map<string, JSONValue>());
+  const itemValues = useRef(propertyValue ?? new Map<string, JSONValue>());
```

Comment:
We shouldn't use propertyValue here since that's a state object that shouldn't be mutated directly (need to use setState for that). Instead, we should construct this map with the propertyValue. Maybe something like:

```
const itemValues = useRef(
  Array.isArray(propertyValue) ? new Map(propertyValue.map(val => [uniqueId(), val])) : 
  new Map<string, JSONValue>());
```

Reply:
Updated. 

>that shouldn't be mutated directly (need to use setState for that

Makes sense, but then why is it okay to set it to a newMap with the same property value?

Reply:
The code never mutates the map values, just the map itself. Before it would have been modifying the actual propertyValue object